keras_cv/metrics/coco/base.py

```diff
@@ -73,6 +73,7 @@ def reset_state(self):
         self.false_positives.assign(tf.zeros_like(self.false_positives))
         self.ground_truth_boxes.assign(tf.zeros_like(self.ground_truth_boxes))
 
+    @tf.function()
```

Comment:
Have you benchmarked the difference with:
https://www.tensorflow.org/api_docs/python/tf/function?version=nightly#using_type_annotations_to_improve_performance

Reply:
Not yet, I'm working on setting up thorough profiling now to figure out the best optimization strategy.

Reply:
I'll be sure to test out this compilation argument - thanks